lib/query-emitter.js

```diff
@@ -311,16 +318,34 @@ QueryEmitter.prototype.onOp = function() {
 };
 
 function getInserted(diff) {
-  var inserted = [];
+  return getDiffItemsByType(diff, arraydiff.InsertDiff)
+}
+
+function getRemoved(diff, values) {
+  const removeDiffs = getDiffItemsByType(diff, arraydiff.RemoveDiff)
+  var removedValues = []
+  for (var i = 0; i < removeDiffs.length; i++) {
+    var { index, howMany } = removeDiffs[i]
+    if (index > -1 && howMany > 0) {
+      removedValues = removedValues.concat(values.slice(index, index + howMany))
+    }
+  }
+  return removedValues
+}
+
+function getDiffItemsByType(diff, diffType) {
```

Comment:
tried to make this generic but then ended up needing specific logic for each diffType. Will likely refactor this so that `getRemoved()` and `getInserted()` don't share any code.